libmamba/include/mamba/specs/repo_data.hpp

```diff
@@ -0,0 +1,238 @@
+        std::size_t build_number = {};
+
+        /** The subdirectory where the package can be found. */
+        std::string subdir = {};
+
+        /** Optionally a MD5 hash of the package archive. */
+        std::optional<std::string> md5 = {};
+
+        /** Optionally a SHA256 hash of the package archive. */
+        std::optional<std::string> sha256 = {};
+
+        /** A deprecated md5 hash. */
+        std::optional<std::string> legacy_bz2_md5 = {};
+
+        /** A deprecated package archive size. */
+        std::optional<std::size_t> legacy_bz2_size = {};
+
+        /** Optionally the size of the package archive in bytes. */
+        std::optional<std::size_t> size = {};
+
+        /** Optionally the architecture the package supports. */
+        std::optional<std::string> arch = {};
+
+        /** Optionally the platform the package supports. */
+        std::optional<std::string> platform = {};
+
+        /** Specification of packages this package depends on. */
+        std::vector<std::string> depends = {};
+
+        /**
+         * Additional constraints on packages.
+         *
+         * `constrains` are different from `depends` in that packages specified in `depends` must
+         * be installed next to this package, whereas packages specified in `constrains` are not
+         * required to be installed, but if they are installed they must follow these constraints.
+         */
+        std::vector<std::string> constrains = {};
+
+        /**
+         * Track features are nowadays only used to deprioritize packages.
+         *
+         * To that effect, the number of track features is counted (number of commas) and the
+         * package is downweighted by the number of track_features.
+         */
+        std::vector<std::string> track_features = {};
+
+        /**
+         * Features are a deprecated way to specify different feature sets for the conda solver.
+         *
+         * This is not supported anymore and should not be used.
+         * Instead, `mutex` packages should be used to specify mutually exclusive features.
+         */
+        std::optional<std::string> features = {};
+
+        /** If this package is independent of architecture this field specifies in what way. */
+        std::optional<NoArchType> noarch = {};
+
+        /** The specific license of the package. */
+        std::optional<std::string> license = {};
+
+        /** The license family. */
+        std::optional<std::string> license_family = {};
+
+        /**
+         * The UNIX Epoch timestamp when this package was created.
+         *
+         * Note that sometimes this is specified in seconds and sometimes in milliseconds.
+         */
+        std::optional<std::size_t> timestamp = {};
+    };
+
+    /**
+     * Serialize to JSON.
+     *
+     * Optional members are omitted from json.
+     */
+    void to_json(nlohmann::json& j, const PackageRecord& p);
+    void to_json(nlohmann::json& j, PackageRecord&& p);
```

Comment:
A const reference can bind to an rvalue, so not sure that we need this overload.